activerecord/lib/active_record/associations.rb

```diff
@@ -1045,45 +1045,45 @@ def association_instance_set(name, association)
       # Indexes are appended for any more successive uses of the table name.
       #
       #   Post.joins(:comments)
-      #   # => SELECT ... FROM posts INNER JOIN comments ON ...
+      #   # SELECT ... FROM posts INNER JOIN comments ON ...
       #   Post.joins(:special_comments) # STI
-      #   # => SELECT ... FROM posts INNER JOIN comments ON ... AND comments.type = 'SpecialComment'
+      #   # SELECT ... FROM posts INNER JOIN comments ON ... AND comments.type = 'SpecialComment'
       #   Post.joins(:comments, :special_comments) # special_comments is the reflection name, posts is the parent table name
-      #   # => SELECT ... FROM posts INNER JOIN comments ON ... INNER JOIN comments special_comments_posts
+      #   # SELECT ... FROM posts INNER JOIN comments ON ... INNER JOIN comments special_comments_posts
```

Comment:
This is only my opinion, but I find the comments more readable with `=>`.  To me, `=>` clearly denotes a postfix comment describing the result of the previous line of code.

Here, for example, how do we know that `SELECT ... FROM posts INNER JOIN comments ON ...` (line 1048) applies to `Post.joins(:comments)` (line 1047) instead of `Post.joins(:special_comments)` (line 1049)?  We might be able to guess because there is no comment on line 1046, but that requires extra thought and still leaves some uncertainty.

Extra line breaks might help here, but I still feel like `=>` is more conventional and communicates unmistakably that "this is the result".

Reply:
I'm also ok with `# =>`, I just prefer to have it be consistent.
Another option could be (although probably difficult to introduce): 
```ruby
# > SELECT ... FROM posts INNER JOIN comments ON ...
# -> SELECT ... FROM posts INNER JOIN comments ON ...
```

Or we could be even more explicit:
```ruby
# generates: SELECT ... FROM posts INNER JOIN comments ON ...
```

As we are on this subject, what do you prefer for raised errors?
```ruby
raise SomeError
# raises SomeError
# or
# => SomeError
```

Reply:
> Another option could be (although probably difficult to introduce):
>
> ```ruby
> # > SELECT ... FROM posts INNER JOIN comments ON ...
> # -> SELECT ... FROM posts INNER JOIN comments ON ...
> ```
>
> Or we could be even more explicit:
>
> ```ruby
> # generates: SELECT ... FROM posts INNER JOIN comments ON ...
> ```

I think `# => ` is a good notation for indicating "result", not just "return value", so my preference would still be `# => `.

But also, for `SELECT` in particular, I think of it like inspecting an object in IRB.  Sometimes we are shown a description (more than just a literal object):

```
irb> Date.new(1999, 12, 31)
=> Fri, 31 Dec 1999
```

Similarly, `SELECT ... FROM posts ...` is a high-level description of the return value.

> As we are on this subject, what do you prefer for raised errors?
>
> ```ruby
> raise SomeError
> # raises SomeError
> # or
> # => SomeError
> ```

I like `# => raises SomeError` (as a "result", not just "return value").  Although we do have `# => SomeError` in many places, and I think it's reasonably understandable.

Reply:
I think rubyists are conditioned to read `=>` as the return statement, so using it for anything (even SQL) seems fine to me. Just that we prefix with a comment, so they aren't parsed as code by the syntax highlighter.

If we have large chunks of return statements, I think splitting them into separate code blocks makes sense.

i.e.

```markdown

\```ruby
foo.blah
\```

Generates:

\```sql
SOME LONG SQL QUERY
\```
```